src/oncall/api/v0/preview.py

```diff
@@ -29,8 +29,19 @@ def on_get(req, resp, schedule_id):
     start__lt = req.get_param('start__lt', required=True)
     end__ge = req.get_param('end__ge', required=True)
     team__eq = req.get_param('team__eq', required=True)
+    roster_id = schedule['roster_id']
 
-    cursor.execute('CREATE TEMPORARY TABLE IF NOT EXISTS `temp_event` AS (SELECT * FROM `event` WHERE `start` < %s AND `end` > %s)', (start__lt, end__ge))
+    # create a temporary table with the events that include members of the team's roster
+    query = '''
```

Comment:
How expensive is this CREATE TABLE? If perf is an issue we can restrict events more finely. I think this should be a relatively small number of events though. Try some sport checking on our prod data.

Reply:
Performance shouldn't be a big problem, the average amount of events copied across teams would be 334 and the max would be 3752; Spot checking on several teams the preview button feels pretty responsive and fast.